projects/templates/src/lib/components/po-page-dynamic-table/po-page-dynamic-table.component.ts

```diff
@@ -436,12 +437,58 @@ export class PoPageDynamicTableComponent extends PoPageDynamicListBaseComponent
     this.navigateTo({ path, params: { duplicate: JSON.stringify(duplicates) } });
   }
 
-  private openEdit(path: string, item) {
+  private openEdit(actionEdit: PoPageDynamicTableActions['edit'], item) {
+    const id = this.formatUniqueKey(item);
+
+    this.subscriptions.add(
+      this.poPageDynamicTableActionsService
+        .beforeEdit(this.actions.beforeEdit, id, item)
+        .pipe(
+          switchMap((beforeEditResult: PoPageDynamicTableBeforeEdit) =>
+            this.executeEditAction(actionEdit, beforeEditResult, item, id)
+          )
+        )
+        .subscribe()
+    );
+  }
+
+  private executeEditAction(
+    action: PoPageDynamicTableActions['edit'],
+    beforeEditResult: PoPageDynamicTableBeforeEdit,
+    item: any,
+    id: string
+  ) {
+    const newEditAction = beforeEditResult?.newUrl ?? action;
+    const allowAction = beforeEditResult?.allowAction ?? true;
+
+    if (!allowAction) {
+      return of({});
+    }
+
+    if (typeof newEditAction === 'string') {
+      this.openEditUrl(newEditAction, item);
+    } else {
+      return of(newEditAction(id, item)).pipe(map(updatedItem => this.modifyUITableItem(item, updatedItem)));
```

Comment:
Nessa parte, por que ele precisa retornar um `of`? Ele não poderia ser assim:
```
const updatedItem = newEditAction(id, item);
this.modifyUITableItem(item, updatedItem);
```
Tem alguma exceção?